From f89d9f31dbae17f999b8f780648360525e0caae6 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 17:33:04 -0400
Subject: [infiniband] qib: Remove duplicate check in get_a_ctxt()

Message-id: <b99f5cd9d12b7f951a7c7d90ce859b858a7e7c63.1411579814.git.dledford@redhat.com>
Patchwork-id: 96694
O-Subject: [Patch RHEL7 11/19] IB/qib: Remove duplicate check in get_a_ctxt()
Bugzilla: 1085961
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: Dan Carpenter <dan.carpenter@oracle.com>

We already know "pusable" is non-zero, no need to check again.

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit db498827ff62611c12c03f6d33bcc532d9fb497e)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/qib/qib_file_ops.c b/drivers/infiniband/hw/qib/qib_file_ops.c
index 2023cd6..b15e34e 100644
--- a/drivers/infiniband/hw/qib/qib_file_ops.c
+++ b/drivers/infiniband/hw/qib/qib_file_ops.c
@@ -1459,7 +1459,7 @@ static int get_a_ctxt(struct file *fp, const struct qib_user_info *uinfo,
      cused++;
     else
      cfree++;
-   if (pusable && cfree && cused < inuse) {
+   if (cfree && cused < inuse) {
     udd = dd;
     inuse = cused;
    }
-- 
1.7.1